subprojects/core/src/main/java/org/gradle/internal/fingerprint/FileCollectionFingerprint.java

```diff
@@ -39,6 +41,12 @@
      */
     Map<String, NormalizedFileSnapshot> getSnapshots();
 
+    /**
+     * The hash of the fingerprint, if available.
+     */
+    @Nullable
+    HashCode getHash();
```

Comment:
How much work would it be to have this as a separate implementation instead? I think we could put in some work to avoid using `null`... Unless we promise to get rid of it soon.